bolt/test/X86/bug-reorder-bb-jrcxz.s

```diff
@@ -0,0 +1,643 @@
+# Test performs a BB reordering with unsupported 
+# instruction jrcxz. Reordering works correctly with the
+# follow options: None, Normal or Reverse. Other strategies 
+# are completed with Assertion `isIntN(Size * 8 + 1, Value).
+# The cause is the distance between BB where one contains 
+# jrcxz instruction.
+# Example: OpenSSL
+# https://github.com/openssl/openssl/blob/master/crypto/bn/asm/x86_64-mont5.pl#L3319
+
+# REQUIRES: system-linux
+
+# RUN: llvm-mc -filetype=obj -triple x86_64-unknown-unknown \
+# RUN:   %s -o %t.o
+# RUN: link_fdata %s %t.o %t.fdata
+# RUN: %host_cc %t.o -falign-labels -march=native -o %t.exe -Wl,-q
+
+# RUN:  llvm-bolt %t.exe -o %t.bolted -data %t.fdata \
+# RUN:    -reorder-blocks=cache+ -reorder-functions=hfsort \
+# RUN:    -split-functions=2 -split-all-cold -split-eh -dyno-stats \
+# RUN:    -print-finalized | FileCheck %s
+
+# CHECK-NOT: "Assertion `isIntN(Size * 8 + 1, Value)"
```

Comment:
In case of an assertion firing, the test will fail before "FileCheck" runs because BOLT will crash.

But that is not an issue because I believe that in the rebased BOLT, when BOLT fails to fix up jrcxz, it does not assert anymore, but it prints an error message and continues processing. Then we need to update this message to match that and verify that we are not hitting this path.

Verify that the test fails without your patch and passes after your patch is applied, i.e., after adding the setIgnored() line to BinaryFunction.cpp, the test starts passing but does not pass without it.

Reply:
done